src/vs/workbench/contrib/chat/browser/chatWidgetService.ts

```diff
@@ -164,6 +162,36 @@ export class ChatWidgetService extends Disposable implements IChatWidgetService
 		return undefined;
 	}
 
+	private async prepareSessionForMove(sessionResource: URI, target: typeof ChatViewPaneTarget | PreferredGroup | undefined): Promise<void> {
+		const existingWidget = this.getWidgetBySessionResource(sessionResource);
+		if (existingWidget) {
+			const existingEditor = isIChatViewViewContext(existingWidget.viewContext) ?
+				undefined :
+				this.editorService.findEditors({ resource: sessionResource, typeId: ChatEditorInput.TypeID, editorId: ChatEditorInput.EditorID }).at(0);
+
+			if (isIChatViewViewContext(existingWidget.viewContext) && target === ChatViewPaneTarget) {
+				return;
+			}
+
+			if (!isIChatViewViewContext(existingWidget.viewContext) && target !== ChatViewPaneTarget && existingEditor && this.isSameEditorTarget(existingEditor.groupId, target)) {
+				return;
+			}
+
+			if (existingEditor) {
+				// widget.clear() on an editor leaves behind an empty chat editor
+				await this.editorService.closeEditor(existingEditor);
```

Comment:
Do we need to check if the editor was closed after all? Isn't it possible that this still brings up a dialog asking for accepting or discarding edits from the session? I am not sure how Connors change is timed for this dialog to disappear?